plugins/vendor/zoom/ZoomDropFolderPlugin/batch/processors/zoomProcessor.php

```diff
@@ -0,0 +1,238 @@
+	const EMAIL = 'email';
+	
+	/**
+	 * @var kZoomClient
+	 */
+	protected $zoomClient;
+	
+	protected $dropFolder;
+	
+	/**
+	 * zoomProcessor constructor.
+	 * @param string $zoomBaseUrl
+	 * @param KalturaZoomDropFolder $folder
+	 */
+	public function __construct($zoomBaseUrl, KalturaZoomDropFolder $folder)
+	{
+		$jwtToken = isset($folder->jwtToken) ? $folder->jwtToken : null;
+		$refreshToken = isset($folder->refreshToken) ? $folder->refreshToken : null;
+		$clientId = isset($folder->clientId) ? $folder->clientId : null;
+		$clientSecret = isset($folder->clientSecret) ? $folder->clientSecret : null;
+		$accessToken = isset($folder->accessToken) ? $folder->accessToken : null;
+		$this->zoomClient = new kZoomClient($zoomBaseUrl, $jwtToken, $refreshToken, $clientId, $clientSecret, $accessToken);
+		$this->dropFolder = $folder;
+	}
+	
+	/**
+	 * @param string $userName
+	 * @return string kalturaUserName
+	 */
+	protected function processZoomUserName($userName)
+	{
+		$result = $userName;
+		switch ($this->dropFolder->zoomVendorIntegration->zoomUserMatchingMode)
+		{
+			case kZoomUsersMatching::DO_NOT_MODIFY:
+				break;
+			case kZoomUsersMatching::ADD_POSTFIX:
+				$postFix = $this->dropFolder->zoomVendorIntegration->zoomUserPostfix;
+				if (!kString::endsWith($result, $postFix, false))
+				{
+					$result = $result . $postFix;
+				}
+				
+				break;
+			case kZoomUsersMatching::REMOVE_POSTFIX:
+				$postFix = $this->dropFolder->zoomVendorIntegration->zoomUserPostfix;
+				if (kString::endsWith($result, $postFix, false))
+				{
+					$result = substr($result, 0, strlen($result) - strlen($postFix));
+				}
+				
+				break;
+			case kZoomUsersMatching::CMS_MATCHING:
+				$zoomUser = $this->zoomClient->retrieveZoomUser($userName);
+				if(isset($zoomUser[self::CMS_USER_FIELD]) && !empty($zoomUser[self::CMS_USER_FIELD]))
+				{
+					$result = $zoomUser[self::CMS_USER_FIELD];
+				}
+				break;
+		}
+		
+		return $result;
+	}
+	
+	/**
+	 * @param $recordingUuId
+	 * @param $recordingPartnerId
+	 * @return KalturaMediaEntry
+	 * @throws PropelException
+	 */
+	protected function getZoomEntryByRecordingId($recordingUuId, $recordingPartnerId)
+	{
+		$entryPager = new KalturaFilterPager();
+		$entryPager->pageSize = 1;
+		$entryPager->pageIndex = 1;
+		
+		$entryFilter = new KalturaBaseEntryFilter();
+		$entryFilter->referenceIdEqual = self::ZOOM_PREFIX . $recordingUuId;
+		
+		try
+		{
+			KBatchBase::impersonate($recordingPartnerId);
+			$kalturaEntry = KBatchBase::$kClient->baseEntry->listAction($entryFilter, $entryPager);
+			KBatchBase::unimpersonate();
+		}
+		catch (Exception $e)
+		{
+			KalturaLog::debug($e->getMessage());
+			$kalturaEntry = null;
+		}
+		
+		if($kalturaEntry->objects)
+		{
+			KalturaLog::debug('Found entry:' . $kalturaEntry->objects[0]->id);
+		}
+		return $kalturaEntry->objects;
+	}
+	
+	/**
+	 * @param string $hostEmail
+	 * @return KalturaUser
+	 */
+	protected function getEntryOwner($hostEmail)

```

Comment:
Few things to fix in this function
1. the only thing that is ever used is the owner ID, so why not just make it getEntryOwnerId(...
2. No really need to check if use exist since if we create entry with user - it will create the user automatically, so just set the correct user ID on the entry based on the Zoom Logic (default, or from cms_user_id ....), no need to check that it exist in Kaltura

Reply:
We also get the partnerId from the owner
on this point the entry didn't created yet, so this is why we first finding/creating the owner by the logic, and only then creating the entry with this owner